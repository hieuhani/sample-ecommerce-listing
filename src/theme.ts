import { DefaultTheme } from 'styled-components'

export const theme: DefaultTheme = {
  colors: {
    reddishOrange: '#f5471e',
    coolGrey: '#8f9598',
    silver: '#c6cace',
    tomato: '#ea341f',
    darkGrey: '#262829',
    paleGrey: '#eef1f3',
    white: '#FFF',
    black30: 'rgba(0, 0, 0, .3)',
    pumpkinOrange: '#f58400',
    deepSkyBlue: '#1e75f6',
  }
}
