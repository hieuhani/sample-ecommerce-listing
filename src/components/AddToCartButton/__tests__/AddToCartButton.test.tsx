import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { AddToCartButton } from '..'
import { theme } from '../../../theme'

const props = {
  cartState: {
    cart: {},
    totalItems: 0,
    totalValue: 0,
  },
  onClick: () => null,
}
describe('AddToCartButton', () => {
  test('AddToCartButton renders correctly', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <AddToCartButton {...props} />
      </ThemeProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('AddToCartButton displays Add To Cart text when totalValue === 0', () => {
    const testProps = props
    const { container } = render(
      <ThemeProvider theme={theme}>
        <AddToCartButton {...testProps} />
      </ThemeProvider>
    )
    expect(container.querySelector('span')).toHaveTextContent('Add To Cart')
  })

  test('AddToCartButton displays cart value when totalValue > 0', () => {
    const testProps = {
      ...props,
      cartState: Object.assign({}, props.cartState, {
        totalValue: 1000,
      })
    }
    const { container } = render(
      <ThemeProvider theme={theme}>
        <AddToCartButton {...testProps} />
      </ThemeProvider>
    )
    expect(container.querySelector('span')).toHaveTextContent('1.000Ä‘')
  })
})
