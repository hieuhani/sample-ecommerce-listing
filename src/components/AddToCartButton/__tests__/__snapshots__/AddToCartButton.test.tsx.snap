// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton AddToCartButton renders correctly 1`] = `
.c0 {
  border: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: rgb(232,32,32);
  background: linear-gradient(90deg,rgba(232,32,32,1) 0%,#f5471e 100%);
  border-radius: 8pt;
  color: #FFF;
  padding: 8pt;
  max-width: 250pt;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 span {
  margin-left: 50pt;
  font-size: 15pt;
}

<button
  className="c0"
  onClick={[Function]}
>
  <img
    src="/static/images/AddToCart.png"
    srcSet="/static/images/AddToCart@2x.png 2x, /static/images/AddToCart@3x.png 3x"
  />
  <span>
    Add To Cart
  </span>
</button>
`;
